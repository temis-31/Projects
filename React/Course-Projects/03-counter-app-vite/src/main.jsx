import React, { StrictMode } from "react";
import ReactDOM from "react-dom/client";
import "./styles.css";

// import { App } from "./HelloWorldApp";
import { FirstApp } from "./FirstApp";
// import CounterApp from "./CounterApp";

ReactDOM.createRoot(document.getElementById("root")).render(
  <StrictMode>
    <FirstApp title="Hello, Im goku" />
  </StrictMode>
);
