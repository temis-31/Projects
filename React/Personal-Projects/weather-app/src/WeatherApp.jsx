import { AddCity } from "./components/AddCity";

export const WeatherApp = () => {

const [city, setCity] = useState("Lima")
  
  const onAddCity = (newCity) => {
  }


  return (
    <>
      <h1>WeatherApp React App</h1>
      <AddCity onNewCity={(event)=> onAddCity(event)}/>

    </>
  );
};
