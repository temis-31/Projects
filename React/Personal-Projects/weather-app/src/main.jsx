import { StrictMode } from "react";
import ReactDOM from "react-dom/client";
import { WeatherApp } from "./WeatherApp";

ReactDOM.createRoot(document.getElementById("root")).render(
  <StrictMode>
    <WeatherApp />
  </StrictMode>
);
