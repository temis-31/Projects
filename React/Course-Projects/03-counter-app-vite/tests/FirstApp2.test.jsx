import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe("Pruebas en <FirstApp />", () => {
  const title = "Hello, Im Goku";
  const subTitle = "Este es un subtitle";

  test("debe hacer match con el snapshot", () => {
    const { container } = render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
  });

  test('debe de mostrar el mensaje "Hello, Im Goku', () => {
    render(<FirstApp title={title} />);
    expect(screen.getByText(title)).toBeTruthy();
    // screen.debug();
  });

  test("debe de mostrar el tÃ­tulo en unh1", () => {
    render(<FirstApp title={title} />);
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      title
    );
  });

  test("debe de mostrar el subtitulo enviado por props", () => {
    render(<FirstApp title={title} subTitle={subTitle} />);
    expect(screen.getAllByText(subTitle).length).toBe(2);
  });
});
