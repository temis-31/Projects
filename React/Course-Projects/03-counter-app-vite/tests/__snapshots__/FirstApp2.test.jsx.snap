// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hello, Im Goku
  </h1>
  <p>
    Hi
  </p>
  <p>
    Hi
  </p>
  <p>
    Hello!!
  </p>
</div>
`;
